cluster/gce/gci/configure-helper.sh

```diff
@@ -66,33 +66,133 @@ function create-dirs {
   fi
 }
 
+function get-local-disk-num() {
+  interface=$1
+  format=$2
+
+  total=0
+  if [[ ! -z ${NODE_LOCAL_SSDS_EXT} ]]; then
+    IFS=";" read -r -a ssdgroups <<< ${NODE_LOCAL_SSDS_EXT}
+    for ssdgroup in "${ssdgroups[@]}"
+    do
+      IFS="," read -r -a ssdopts <<< ${ssdgroup}
+      opnum=${ssdopts[0]}
+      opinterface=${ssdopts[1]}
+      opformat=${ssdopts[2]}
+
+      if [[ ${opformat,,} == ${format,,} && ${opinterface,,} == ${interface,,} ]]; then
+        total=$((total+opnum))
+      fi
+    done
+  fi
+  
+}
+
+function safe-block-symlink(){
+  device=$1
+  interface=$2
+  mkdir -p "/mnt/disks/by-uuid/google-local-ssds-${interface,,}-block"
+  ssdmap="/home/kubernetes/localssdmap.txt"
+  echo "Creating symlink for block devices..."
+
+  if [[ ! -e ${ssdmap} ]]; then
+    touch ${ssdmap}
+    chmod +w ${ssdmap}
+  fi
+
+  # each line looks like "${device} persistant-uuid"
+  if [[ ! -z $(cat ${ssdmap} | grep ${device}) ]]; then
+    #create symlink based on saved uuid
+    myuuid=$(cat ${ssdmap} | grep ${device} | cut -d ' ' -f 2)
+  else
+    # generate new uuid and add it to the map
+    myuuid=$(uuidgen)
+    echo "${device} ${myuuid}" >> ${ssdmap}
+  fi
+
+  sym="/mnt/disks/by-uuid/google-local-ssds-${interface,,}-block/local-ssd-${myuuid}"
+  # dont make the exact directory first or else symlink goes 1 directory down
+  ln -s ${device} ${sym}
+  echo "Created a symlink for SSD $ssd at ${sym}"
+  chmod a+w ${sym}
+}
+
+
 # Formats the given device ($1) if needed and mounts it at given mount point
 # ($2).
-function safe-format-and-mount() {
+function safe-format-and-mount-by-interface() {
   device=$1
-  mountpoint=$2
+  interface=$2
 
   # Format only if the disk is not already formatted.
   if ! tune2fs -l "${device}" ; then
     echo "Formatting '${device}'"
     mkfs.ext4 -F "${device}"
   fi
 
+  # Trigger udev refresh so that newly formatted devices are propagated in by-uuid
+  udevadm control --reload-rules
+  udevadm trigger
+  udevadm settle
+  
+  # Get the real UUID of the device
+  item=$(readlink -f ${device} | cut -d '/' -f 3)
+  myuuid=$(ls -l /dev/disk/by-uuid/ | grep ${item} | tr -s ' ' | cut -d ' ' -f 9) 
+  mountpoint="/mnt/disks/by-uuid/google-local-ssds-${interface,,}-fs/local-ssd-${myuuid}"
+
+  # Mount device to the mountpoint
   mkdir -p "${mountpoint}"
   echo "Mounting '${device}' at '${mountpoint}'"
   mount -o discard,defaults "${device}" "${mountpoint}"
+  chmod a+w ${mountpoint}
```

Comment:
Do we need to do this considering that it wasn't done previously?

Reply:
I believe this was done previously on a mount: https://github.com/kubernetes/kubernetes/blob/fa916c1002992ab0e7e6e044c68b0e52c4ef0a50/cluster/gce/gci/configure-helper.sh#L95